src/infrasys/time_series_models.py

```diff
@@ -89,6 +101,72 @@ def check_data(
 
         return data
 
+    @classmethod
+    def aggregate(cls, ts_data: list[Self]) -> Self:
+        """Method to aggregate list of SingleTimeSeries data.
+
+        Parameters
+        ----------
+        ts_data
+            List of SingleTimeSeries data
+
+        Returns
+        -------
+        SingleTimeSeries
+
+        Raises
+        ------
+        InconsistentTimeseriesAggregation
+            Raised if incompatible timeseries data are passed.
+        """
+
+        # Extract unique properties from ts_data
+        unique_props = {
+            "length": {data.length for data in ts_data},
+            "resolution": {data.resolution for data in ts_data},
+            "start_time": {data.initial_time for data in ts_data},
+            "variable": {data.variable_name for data in ts_data},
+            "data_type": {type(data.data) for data in ts_data},
+        }
+
+        # Validate uniformity across properties
+        if any(len(prop) != 1 for prop in unique_props.values()):
+            msg = f"Inconsistent timeseries data: {unique_props}"
```

Comment:
Perhaps this should only print the sets with length greater than one.

Reply:
Sure